lib/preload.js

```diff
@@ -382,13 +382,56 @@ class Preloader extends EventEmitter {
 			.catch((error) => this.emit('error', error));
 	}
 
+	/**
+	 * Send a command to `preload.py` (running in a Docker container) through
+	 * its stdin, and read a response from its stdout. The stdout response
+	 * must be a JSON object containing a `statusCode` field (integer) and
+	 * optionally `result` and `error` fields. If `statusCode` is the number
+	 * zero, the command execution is considered successful, otherwise it is
+	 * assumed to have failed and the returned promise is rejected with an
+	 * error message including the message provided in the `error` field.
+	 *
+	 * @param {string} command
+	 * @param {object} parameters
+	 */
 	_runCommand(command, parameters) {
 		return new Bluebird((resolve, reject) => {
+			const cmd = JSON.stringify({ command, parameters }) + '\n';
 			this.stdout.once('error', reject);
 			this.stdout.once('data', (data) => {
-				resolve(JSON.parse(data).result);
+				let strData = data;
+				try {
+					strData = data.toString();
```

Comment:
`data` is normally a `Buffer` object, not a string. `JSON.parse(data)` would implicitly call `data.toString()`, now I'm making it explicit so it can also be used in error messages.